### PR TITLE
Fix slow point deletion in RemovePoints.

### PR DESCRIPTION
**Pull request by SirVer**
This speeds up the asset writer significantly:

For the 3D sample bag:

~~~
/usr/bin/time roslaunch cartographer_ros assets_writer_backpack_3d.launch \
   bag_filenames:=${HOME}/Downloads/b3-2016-04-05-14-14-00.bag \
   pose_graph_filename:=${HOME}/Downloads/b3-2016-04-05-14-14-00.bag.pbstream

master:
2100.96user 26.79system 35:43.60elapsed 99%CPU (0avgtext+0avgdata 3657028maxresident)k
17682632inputs+32080outputs (0major+20021051minor)pagefaults 0swaps

this CL:
1937.46user 11.77system 32:22.49elapsed 100%CPU (0avgtext+0avgdata 3659412maxresident)k
1352inputs+32088outputs (0major+7042032minor)pagefaults 0swaps
~~~

So ~10%. For pipelines that do less this can be even 30% or more.

### CODE REVIEW

**Comment by SirVer** [comment]
Benchmarks are coming monday.

File: cartographer/io/points_batch.cc [hunk 1]
```diff
@@ -19,17 +19,29 @@
 namespace cartographer {
 namespace io {
 
-void RemovePoints(std::vector<int> to_remove, PointsBatch* batch) {
-  std::sort(to_remove.begin(), to_remove.end(), std::greater<int>());
-  for (const int index : to_remove) {
-    batch->points.erase(batch->points.begin() + index);
+void RemovePoints(std::unordered_set<int> to_remove, PointsBatch* batch) {
+  std::vector<Eigen::Vector3f> points;
+  points.reserve(batch->points.size());
```
Comment (cschuet) [inline cartographer/io/points_batch.cc]: Can't you just reserve batch->points.size() - to_remove.size()) here?

File: cartographer/io/points_batch.cc [hunk 2]
```diff
@@ -19,17 +19,29 @@
...
+  std::vector<float> intensities;
+  intensities.reserve(batch->intensities.size());
```
Comment (cschuet) [inline cartographer/io/points_batch.cc]: here is guess you have to check and only reserve batch->intensities.size() - to_remove.size()) if batch->intensities.size() > 0

**Comment by SirVer** [comment]
@cschuet PTAL.

**Pull request merged**